corehq/apps/accounting/tests/test_migrations.py

```diff
@@ -0,0 +1,135 @@
+import random
+from datetime import date, timedelta
+
+from django.test import TransactionTestCase
+
+from corehq.apps.accounting.tests import generator
+from corehq.apps.accounting.models import (
+    BillingAccount,
+    DefaultProductPlan,
+    SoftwarePlanEdition,
+    Subscription,
+)
+from corehq.apps.accounting.tasks import assign_explicit_community_subscriptions
+from corehq.apps.domain.models import Domain
+
+
+class TestExplicitCommunitySubscriptions(TransactionTestCase):
+
+    domain = None
+    from_date = None
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestExplicitCommunitySubscriptions, cls).setUpClass()
+
+        cls.domain = Domain(name='test')
+        cls.domain.save()
+        cls.from_date = date.today()
+
+    def setUp(self):
+        super(TestExplicitCommunitySubscriptions, self).setUp()
+        generator.instantiate_accounting()
+        assert Subscription.objects.count() == 0
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.domain.delete()
+        super(TestExplicitCommunitySubscriptions, cls).tearDownClass()
+
+    def tearDown(self):
+        generator.delete_all_subscriptions()
+        generator.delete_all_accounts()
+        super(TestExplicitCommunitySubscriptions, self).tearDown()
+
+    def test_no_preexisting_subscription(self):
+        self._assign_community_subscriptions()
+
+        self.assertEqual(Subscription.objects.count(), 1)
+        subscription = Subscription.objects.all()[0]
+        self.assertEqual(subscription.subscriber.domain, self.domain.name)
+        self.assertEqual(subscription.date_start, self.from_date)
+        self.assertIsNone(subscription.date_end)
+        self.assertEqual(subscription.plan_version, self._most_recently_created_community_plan_version)
+
+    def test_preexisting_current_subscription(self):
+        preexisting_subscription = Subscription.new_domain_subscription(
+            self._preexisting_subscription_account,
+            self.domain.name,
+            self._random_plan_version,
+        )
+
+        self._assign_community_subscriptions()
+
+        self.assertEqual(Subscription.objects.count(), 1)
+        self.assertFalse(Subscription.objects.exclude(subscriber__domain=self.domain.name).exists())
+        self.assertEqual(Subscription.objects.all()[0], preexisting_subscription)
+
+    def test_preexisting_future_subscription(self):
+        future_subscription_start_date = self.from_date + timedelta(days=10)
+        plan_version = self._random_plan_version
+        Subscription.new_domain_subscription(
+            self._preexisting_subscription_account,
+            self.domain.name,
+            plan_version,
+            date_start=future_subscription_start_date,
+        )
+
+        self._assign_community_subscriptions()
+
+        self.assertEqual(Subscription.objects.count(), 2)
```

Comment:
> Is the constant 2 here dependent on the previous test creating a subscription?

No, I create two subscriptions during the run of this test, and make sure they're there.  There's no dependency between the individual tests